libs/authentication/api/src/lib/authentication.service.ts

```diff
@@ -0,0 +1,79 @@
+import {Injectable} from '@nestjs/common'
+import {PrismaClient} from '@prisma/client'
+import {AuthSessionType, AuthSession} from './auth-session'
+import {unselectPassword} from '@wepublish/user/api'
+
+@Injectable()
+export class AuthenticationService {
+  constructor(private prisma: PrismaClient) {}
+
+  public async getSessionByToken(token: string): Promise<AuthSession | null> {
+    const [tokenMatch, session] = await Promise.all([
+      this.prisma.token.findFirst({
+        where: {
+          token
+        }
+      }),
+      this.prisma.session.findFirst({
+        where: {
+          token
+        }
+      })
+    ])
+
+    if (tokenMatch) {
+      return {
+        type: AuthSessionType.Token,
+        id: tokenMatch.id,
+        name: tokenMatch.name,
+        token: tokenMatch.token,
+        roles: await this.prisma.userRole.findMany({
+          where: {
+            id: {
+              in: tokenMatch.roleIDs
+            }
+          }
+        })
+      }
+    }
+
+    if (session) {
+      const user = await this.prisma.user.findUnique({
+        where: {
+          id: session.userID
+        },
+        select: unselectPassword
+      })
+
+      if (!user) {
+        return null
+      }
+
+      return {
+        type: AuthSessionType.User,
+        id: session.id,
+        token: session.token,
+        createdAt: session.createdAt,
+        expiresAt: session.expiresAt,
+        user,
+        roles: await this.prisma.userRole.findMany({
+          where: {
+            id: {
+              in: user.roleIDs
+            }
+          }
+        })
+      }
+    }
+
+    return null
+  }
+
+  public isSessionValid(session: AuthSession | null) {
+    if (!session) {
+      return false
+    }
+
+    return session.type === 'user' ? session.expiresAt > new Date() : true
```

Comment:
Use `AuthSessionType.User`

I am not huge fan of this tennary.

Rather use
```
return session.type === AuthSessionType.User && session.expiredAt <= new Date();
```
?

Reply:
Oh yeah missed that, switched it to a string when I didn't fix the circular dependency before